src/web/components/elements/InteractiveBlockComponent.tsx

```diff
@@ -89,17 +248,36 @@ export const InteractiveBlockComponent = ({ url, scriptUrl, alt }: Props) => {
 					}
 				},
 			);
+
+			setLoaded(true);
 		}
-	}, [scriptUrl]);
+	}, [scriptUrl, role, url, loaded]);
 
 	return (
 		<div
 			data-cypress={`interactive-element-${encodeURI(alt || '')}`}
 			ref={wrapperRef}
+			className={wrapperStyle(role, loaded)}
 		>
-			<a data-name="placeholder" href={url}>
-				{alt}
-			</a>
+			{!loaded && (
+				<Placeholder
+					height={
+						role === 'supporting'
+							? supportingMinHeight
+							: inlineMinHeight
+					}
+				/>
+			)}
+			{!loaded && (
+				<a
+					ref={placeholderLinkRef}
+					data-name="placeholder"
+					className={placeholderLinkStyle}
+					href={url}
+				>
+					{alt}
+				</a>
+			)}
```

Comment:
🤔 I'm only half understanding this code but would it work if we server rendered the placeholder and then used a portal to replace it with the final content?

Or. Is it that we have 2 loading states. The SSR placeholder, _and_ also a client side placeholder for when we're making the api call to get boot.js? In which case... I think youo could have a more classic setup where you use `useApi` to fetch the script and render a placeholder based on the `loading` prop that this custom hook returns?

I feel like I'm missing an aspect to this though so I could be oversimplifying things. I can see there's something around needing to know the height of something after it's rendered...?

Sorry, that was a very rambling comment!

Reply:
There's a few difficulties:

- The script is an AMD module so we need to use an AMD loader to fetch it, we can't use `useAPI`
- I tried to use portal and it just added a new div element in the container and I wasn't sure why, hydration works as expected (I am 'hydrating' the placeholder and anchor with the client side fetching stuff essentially, there's not an empty `div` like portal expects)
- The steps are placeholder -> setup listeners -> switch out for iframe -> iframe calls out to. parent and sets its height. So there's a contract there that needs to be taken into account.

Reply:
>     * added a new div element in the container

I think Portal is looking for a marker so it can remove the placeholder... This:

```
	const placeholderElement = element.querySelector(
		'[data-name="placeholder"]',
	);
```

Reply:
This is the problem with weird contracts that are hard to discover... 😞

Reply:
Yeah, that's in there from where it was a portal before 🤔

Reply:
I'm going to leave this as is because I think I *want* server-rendered placeholder and if I was to use a portal I'd need to branch the interactiveBlockElement into a server-rendered element and client rendered element, whereas I can just handle that in the component itself.

Reply:
I'll catch up with you directly. As far as I'm aware the Portal is designed for this purpose. You have a SSR placeholder and then that gets replaced by the actual content on the client following an api call.

Sorry, I've been tied with lots of things today so I haven't been able to dig into tis more (or respond sooner!)

Reply:
Yeah, no rush, we can look next week, I'd be keen to get to the bottom of the pattern and happy to update if it makes sense!